vm/devices/get/get_protocol/src/lib.rs

```diff
@@ -611,6 +609,44 @@ pub struct IgvmAttestRequest {
 
 const_assert_eq!(6288, size_of::<IgvmAttestRequest>());
 
+open_enum! {
+    /// This enum is the same as `IgvmAttestRequestType` defined in `underhill_attestation`.
+    /// Have a copy of the enum to avoid adding additional dependency. The enum is used by
```

Comment:
if you're using this for test only, I would move these declarations to the test code using it.

(or just take a cross-cutting dependency on `underhill_attestation` in test-only, via `dev-dependencies` in Cargo.toml - assuming there aren't circular dep issues)

Reply:
Is GED used by vmm tests guarded by tests?

Reply:
Ah, I understand your question.

My suggestion would be to split the `underhill_attestation::protocol` module into its own crate, and have the GED import _that_ crate.

Pulling in the entire `underhill_attestation` crate, with its myriad of dependencies and business logic, likely wouldn't work (or if it did work, severely bloat OpenVMM's compile times for not much benefit)

Reply:
there is a circular dependency 😢

Reply:
I think I can workaround it

Reply:
From my grepping of the code, it seems that circular dep is one you just introduced in this PR?

If so, I might suggest breaking that circular dep by sidestepping the use of `get_protocol` constants in things like `WRAPPED_KEY_RESPONSE_BUFFER_PAGES`, etc...

Reply:
After all - from a layering POV, its not really the GET protocol that dictates the size of these responses, right? Its the responses themselves that have a certain size, and the GET needs to handle whatever sizes they happen to be?

Reply:
yeah, I'm thinking about the same.

Reply:
> After all - from a layering POV, its not really the GET protocol that dictates the size of these responses, right? Its the responses themselves that have a certain size, and the GET needs to handle whatever sizes they happen to be?

The GET will specify the buffer size in the request and the GED will write the response to the buffer upto the buffer size

Reply:
But only the maximum is the contract. The actual value set by the request is up to the caller to decide.

Reply:
Let me ping you offline to better understand this